### PR TITLE
CONCF-978 restrict permissions to getData and setData to admins only

### PR DESCRIPTION
**Pull request by bognix**
related PR: https://github.com/Wikia/mercury/pull/1215

/cc @hakubo @rogatty

### CODE REVIEW

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 1]
```diff
@@ -483,48 +485,54 @@ public function setData( ) {
 					wfRunHooks( 'CuratedContentSave', [ $sections ] );
 				}
 			}
+			$this->response->setVal( 'status', $status );
 
+		} else {
+			$this->response->setCode( \Wikia\Service\ForbiddenException::CODE );
+			$this->response->setVal( 'message', 'No permissions to save curated content' );
```
Comment (vforge) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php:492]: Question: Why you're returning message if you aren't reading it on front-end side?
/edit Talked offline - not needed, but added for extra debugging/consistency purposes.

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 2]
```diff
@@ -483,48 +485,54 @@ public function setData( ) {
...
-		}
 
-		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
-		$this->response->setVal( 'data', $data );
-		// TODO: CONCF-961 Set more restrictive header
-		$this->response->setHeader( 'Access-Control-Allow-Origin', '*' );
+			$this->response->setVal( 'data', $data );
+		} else {
+			$this->response->setCode( \Wikia\Service\ForbiddenException::CODE );
+			$this->response->setVal( 'message', 'No permissions to access curated content' );
```
Comment (vforge) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php:534]: Question: Why you're returning message if you aren't reading it on front-end side?
/edit Talked offline - not needed, but added for extra debugging/consistency purposes.

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 3]
```diff
@@ -483,48 +485,54 @@ public function setData( ) {
...
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
 		// TODO: CONCF-961 Set more restrictive header
 		$this->response->setHeader( 'Access-Control-Allow-Origin', '*' );
-	}
 
-	public function getData( ) {
-		global $wgWikiaCuratedContent;
-		$data = [];
+		if ( $wgUser->isAllowed( 'curatedcontent' ) ) {
+			$data = [];
```
Comment (vforge) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php:504]: Missing space.

**Comment by vforge** [comment]
Minor styling, LGTM beside that.

**Comment by hakubo** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**